### PR TITLE
Fixed SQL injections by moving from queries to prepared statements

### PR DESCRIPTION
**Pull request by AlexGustafsson**
I've fixed the SQL injections (see #1).

### CODE REVIEW

File: src/lr/lr.php [hunk 1]
```diff
@@ -4,23 +4,42 @@
 	header("Location: ../login.php");
 }
 require_once("../php/connect.php");
-$usernow = mysql_query("SELECT * FROM `user` WHERE `id` = '{$_SESSION['user']}'");
+$stmt = $conn->prepare("SELECT * FROM `user` WHERE `id` = ?");
+$stmt->bind_param("i", $_SESSION['user']);
+$stmt->execute();
+$usernow = $stmt->get_results();
```
Comment (sunboyy) [inline src/lr/lr.php]: `get_results` does not work, please consider replacing with `get_result` instead

**Review by sunboyy — CHANGES_REQUESTED** [review]
It seems that mysqli_stmt object doesn't have `get_results` method. Please use `get_result` instead

File: src/lr/lr.php (hunk 1 above)
Comment (AlexGustafsson) [reply src/lr/lr.php]: Solved in the latest commit.

File: src/recallcard/randomcard.php [hunk 2]
```diff
@@ -22,17 +22,31 @@
 }
 if ($type == "lesson") {
 	$lesson = get_lesson($getid);
-	$group = $conn->query("SELECT * FROM `group` WHERE `id` = '{$data_lesson['group']}'");
+	$stmt = $conn->prepare("SELECT * FROM `group` WHERE `id` = ?");
+	$stmt->bind_param("i", $data_lesson['group']);
+	$stmt->execute();
+	$thisgroup = $stmt->get_result();
+	$data_thisgroup = $thisgroup->fetch_array();
```
Comment (sunboyy) [inline src/recallcard/randomcard.php]: There is a duplicate `->fetch_array()` (this line and the following line)

File: src/recallcard/recallcard.php [hunk 3]
```diff
@@ -70,7 +78,12 @@
 			<td>
 			  <div id="mainright">
 				<?php foreach (get_groups() as $data_group) {
-				$lesson = $conn->query("SELECT * FROM `lesson` WHERE `group` = '{$data_group['id']}' ORDER BY `id` ASC");
+				$stmt = $conn->prepare("SELECT * FROM `lesson` WHERE `group` = ? ORDER BY `id` ASC");
+				$stmt->bind_param("i", $data_group['id']);
+				$stmt->execute();
+				$lesson = $stmt->get_result();
+				$data_checkuser = $checkuser->fetch_array();
```
Comment (sunboyy) [inline src/recallcard/recallcard.php]: This variable is not needed.

File: src/recallcard/randomcard.php (hunk 2 above)
Comment (sunboyy) [reply src/recallcard/randomcard.php]: You can replace this line and the line above with `$group = $stmt->get_result();` and the following line will continue doing its job

File: src/recallcard/core.php [hunk 4]
```diff
@@ -24,15 +26,19 @@ function get_cards($type, $value) {
...
+        $qry_cards = $stmt->get_result();
         while ($row_cards = $qry_cards->fetch_array()) {
             array_push($cards, $row_cards);
         }
     }
     else if ($type == "lesson") {
-        $sql_cards = "SELECT * FROM `card` WHERE `lesson` = $value";
-        $qry_cards = $conn->query($sql_cards);
+        $stmt = $conn->prepare("SELECT * FROM `card` WHERE `lesson` = ?");
+        $stmt->bind_param("s", $value);
```
Comment (sunboyy) [inline src/recallcard/core.php]: `lesson` is integer, so use "i" instead

File: src/recallcard/randomcard.php [hunk 5]
```diff
@@ -22,17 +22,31 @@
...
+
+	$stmt = $conn->prepare("SELECT * FROM `lesson` WHERE `group` = ?");
+	$stmt->bind_param("i", $getid);
+	$stmt->execute();
+	$lesson = $stmt->get_result();
 	$num_lesson = $lesson->num_rows;
 }
 else if ($type = "selected") {
-	$group = $conn->query("SELECT * FROM `group` WHERE `id` = '$getid'");
+	$stmt = $conn->prepare("SELECT * FROM `group` WHERE `id` = '$getid'");
```
Comment (sunboyy) [inline src/recallcard/randomcard.php]: Use `?`

File: src/recallcard/recallcard_add.php [hunk 6]
```diff
@@ -9,7 +9,9 @@
 	$lsnname = htmlspecialchars(trim($_POST['lsnname']));
 	$lsngroup = $_POST['lsngroup'];
 	if (($lsnname != "") && ($lsngroup != "")) {
-		$conn->query("INSERT INTO `lesson` VALUES ('', '$lsnname', '$lsngroup', '{$user['id']}', NOW())");
+		$stmt = $conn->prepare("INSERT INTO `lesson` VALUES ('', ?, ?, ?, NOW())");
+		$stmt->bind_param("ssi", $lsnname, $lsngroup, $user['id']);
```
Comment (sunboyy) [inline src/recallcard/recallcard_add.php]: `sii`

**Review by sunboyy — CHANGES_REQUESTED** [review]

File: src/recallcard/recallcard_add.php (hunk 6 above)
Comment (AlexGustafsson) [reply src/recallcard/recallcard_add.php]: Done.

File: src/recallcard/randomcard.php (hunk 5 above)
Comment (AlexGustafsson) [reply src/recallcard/randomcard.php]: Done.

File: src/recallcard/core.php (hunk 4 above)
Comment (AlexGustafsson) [reply src/recallcard/core.php]: Done.

File: src/recallcard/recallcard.php (hunk 3 above)
Comment (AlexGustafsson) [reply src/recallcard/recallcard.php]: Fixed.

File: src/recallcard/randomcard.php (hunk 2 above)
Comment (AlexGustafsson) [reply src/recallcard/randomcard.php]: Done.

Comment (sunboyy) [reply src/recallcard/randomcard.php]: It seems that `$data_group` variable disappears

File: src/recallcard/randomcard.php [hunk 7]
```diff
@@ -22,17 +22,29 @@
 }
 if ($type == "lesson") {
 	$lesson = get_lesson($getid);
-	$group = $conn->query("SELECT * FROM `group` WHERE `id` = '{$data_lesson['group']}'");
-	$data_group = $group->fetch_array();
+	$stmt = $conn->prepare("SELECT * FROM `group` WHERE `id` = ?");
+	$stmt->bind_param("i", $data_lesson['group']);
+	$stmt->execute();
+	$group = $stmt->get_result();
```
Comment (sunboyy) [inline src/recallcard/randomcard.php:28]: It looks like `$data_group` variable disappears. Could you bring it back?

Comment (AlexGustafsson) [reply src/recallcard/randomcard.php:28]: Fixed.

**Review by sunboyy — APPROVED** [review]

**Pull request merged**